scripts/conversion_script.py

```diff
@@ -0,0 +1,477 @@
+                os.makedirs(directory, mode=0o755)
+
+            print("Writing file to destination: '%s'" % required_file.path)
+            with open(required_file.path, mode="w") as handler:
+                handler.write(data)
+                os.chmod(required_file.path, 0o644)
+
+
+# Code taken from
+# https://github.com/oamg/convert2rhel/blob/v1.4.1/convert2rhel/utils.py#L345
+# and modified to adapt the needs of the tools that are being executed in this
+# script.
+def run_subprocess(cmd, print_cmd=True, env=None):
+    """
+    Call the passed command and optionally log the called command
+    (print_cmd=True) and environment variables in form of dictionary(env=None).
+    Switching off printing the command can be useful in case it contains a
+    password in plain text.
+
+    The cmd is specified as a list starting with the command and followed by a
+    list of arguments. Example: ["yum", "install", "<package>"]
+    """
+    # This check is here because we passed in strings in the past and changed
+    # to a list for security hardening.  Remove this once everyone is
+    # comfortable with using a list instead.
+    if isinstance(cmd, str):
+        raise TypeError("cmd should be a list, not a str")
+
+    if print_cmd:
+        print("Calling command '%s'" % " ".join(cmd))
+
+    process = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, bufsize=1, env=env
+    )
+    output = ""
+    for line in iter(process.stdout.readline, b""):
+        line = line.decode("utf8")
+        output += line
+
+    # Call wait() to wait for the process to terminate so that we can
+    # get the return code.
+    process.wait()
+
+    return output, process.returncode
+
+
+def install_convert2rhel():
+    """Install the convert2rhel tool to the system."""
+    print("Installing & updating Convert2RHEL package.")
+    output, returncode = run_subprocess(
+        ["yum", "install", "convert2rhel", "-y"],
+    )
+    if returncode:
+        raise ProcessError(
+            message="Failed to install convert2rhel RPM.",
+            report="Installing convert2rhel with yum exited with code '%s' and output: %s."
+            % (returncode, output.rstrip("\n")),
+        )
+
+    output, returncode = run_subprocess(["yum", "update", "convert2rhel", "-y"])
+    if returncode:
+        raise ProcessError(
+            message="Failed to update convert2rhel RPM.",
+            report="Updating convert2rhel with yum exited with code '%s' and output: %s."
+            % (returncode, output.rstrip("\n")),
+        )
+
+
+def run_convert2rhel():
+    """
+    Run the convert2rhel tool assigning the correct environment variables.
+    """
+    print("Running Convert2RHEL Analysis")
+    env = {"PATH": os.environ["PATH"]}
+
+    if "RHC_WORKER_CONVERT2RHEL_DISABLE_TELEMETRY" in os.environ:
+        env["CONVERT2RHEL_DISABLE_TELEMETRY"] = os.environ[
+            "RHC_WORKER_CONVERT2RHEL_DISABLE_TELEMETRY"
+        ]
```

Comment:
Currently, not right now... This is some leftover from the analysis script. Since we don't support any type of variables from insights ui to be passed down to the script.

Reply:
Alright :+1: